torch/nn/parallel/scatter_gather.py

```diff
@@ -10,16 +10,19 @@ def scatter(inputs, target_gpus, dim=0):
     references to objects that are not variables. Does not
     support Tensors.
     """
+    def scatter_var(obj):
+        return Scatter.apply(target_gpus, None, dim, obj)
+
     def scatter_map(obj):
         if isinstance(obj, Variable):
-            return Scatter.apply(target_gpus, None, dim, obj)
+            return scatter_var(obj)
         assert not torch.is_tensor(obj), "Tensors not supported in scatter."
         if isinstance(obj, tuple) and len(obj) > 0:
-            return list(zip(*map(scatter_map, obj)))
+            return list(zip(*map(scatter_var, obj)))
         if isinstance(obj, list) and len(obj) > 0:
-            return list(map(list, zip(*map(scatter_map, obj))))
+            return list(map(list, zip(*map(scatter_var, obj))))
         if isinstance(obj, dict) and len(obj) > 0:
-            return list(map(type(obj), zip(*map(scatter_map, obj.items()))))
+            return list(map(type(obj), zip(*map(scatter_var, obj.items()))))
```

Comment:
But after this change the function will no longer be able to handle arbitrarily nested data structures, right? I don't think it's a good fix.